clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -52,22 +70,33 @@ public AbstractRequestResponse getErrorResponse(int versionId, Throwable e) {
         Errors error = Errors.forException(e);
         List<MetadataResponse.PartitionMetadata> partitions = Collections.emptyList();
 
-        for (String topic : topics)
-            topicMetadatas.add(new MetadataResponse.TopicMetadata(error, topic, partitions));
+        if (topics != null) {
+            for (String topic : topics)
+                topicMetadatas.add(new MetadataResponse.TopicMetadata(error, topic, false, partitions));
+        }
 
         switch (versionId) {
             case 0:
-                return new MetadataResponse(Collections.<Node>emptyList(), topicMetadatas);
+            case 1:
```

Comment:
This can fall through because the constructor supports all the versions. The version is passed as the last parameter. 

This is related to the discussion here: https://github.com/apache/kafka/pull/1095#discussion_r60758124